### ISSUE TITLE
Issue with Code Execution Results

### ISSUE THREAD

**Issue by airoue** [Meizhen2023/SARID#5]
Hello, and thank you for making this code available!
I am trying to replicate the results from your paper,  but I encountered an issue. Following the instructions, I used the data_processing module to extract MFCC features and then ran the Transformers model. However, the R2 score I obtain during testing is negative.The best result is as follows:
![myplot](https://github.com/user-attachments/assets/88b902be-88a7-46e2-981d-80c497b38eb6)

Could you provide any insights or recommendations on specific parameters, preprocessing steps, or settings for feature extraction and model training that might affect the R2 outcome? Any guidance on potential areas to review would be greatly appreciated.
Thank you in advance for your time and help!

**Comment by Meizhen2023** [comment]
Hi, something I can suggest is to check every step of your process, feature, model, label, prediction, etc.

**Comment by Meizhen2023** [comment]
If you make sure that your model is correct, please pay attention to your feature extraction, label-predictions matching. Or you could first use train dataset as test dataset to test if your data processing is correct.

**Comment by airoue** [comment]
Thank you for your reply and valuable suggestions! Based on your advice, I have carried out the following operations:
Feature Extraction:
In data_processing.py, I used the following code to extract the MFCC features:
`X_mfcc, Y_mfcc = audio_utils.preprocessing(audio_utils.dataset_path_{train/test})
Y_mfcc.to_csv('{train/test}_label_nmfcc400.csv')
np.save("{train/test}_mfcc_nmfcc400", X_mfcc)`
The extracted data has been saved in the working directory.

Use train dataset as test dataset to test:
In baseline.py, I made the following modifications:
Changed line 179 to:
`acoustic_feature = train_dataset.feature.transpose(0, 2, 1)`
Changed line 195 to:
`labels = train_dataset.label['RAINFALL INTENSITY'].to_numpy()`
However, despite these changes, the R2 score remains negative.
